include/SampleBuffer.h

```diff
@@ -269,6 +272,11 @@ class LMMS_EXPORT SampleBuffer : public QObject, public sharedObject
 	}
 
 
+	typedef std::array<std::array<sample_t, OscillatorConstants::WAVETABLE_LENGTH>,
+		OscillatorConstants::WAVE_TABLES_PER_WAVEFORM_COUNT> wavetable_t;
```

Comment:
Which file should the `typedef` belong to? Also, can we have another `typedef` for plain array?

Reply:
Maybe it could be moved to `Oscillator.h` or `OscillatorConstants.h`, since it already depends on some constants from it. By another `typedef` for plain array I'm assuming you mean 1D `std::array` (not plain "C-style array"). Makes sense, it could make the construction more readable.

Reply:
Moved to `OscillatorConstants.h` and renamed based on what the length constants referred to:
```c
typedef std::array<sample_t, OscillatorConstants::WAVETABLE_LENGTH> wavetable_t;
typedef std::array<wavetable_t, OscillatorConstants::WAVE_TABLES_PER_WAVEFORM_COUNT> waveform_t;
```